pkg/vpwned/server/vjailbreak_proxy.go

```diff
@@ -236,3 +237,43 @@ func CreateInClusterClient() (client.Client, error) {
 	}
 	return clientset, nil
 }
+
+func (p *vjailbreakProxy) RevalidateCredentials(ctx context.Context, in *api.RevalidateCredentialsRequest) (*api.RevalidateCredentialsResponse, error) {
+    kind := in.GetKind()
+    name := in.GetName()
+    namespace := in.GetNamespace()
+
+    switch kind {
+
+    case "VmwareCreds":
+        vmwcreds := &vjailbreakv1alpha1.VMwareCreds{}
+        if err := p.K8sClient.Get(ctx, k8stypes.NamespacedName{Name: name, Namespace: namespace}, vmwcreds); err != nil {
+            return nil, fmt.Errorf("failed to get VMwareCreds %s/%s: %w", namespace, name, err)
+        }
+
+        vmwcreds.Status.VMwareValidationStatus = ""
+        vmwcreds.Status.VMwareValidationMessage = ""
+        if err := p.K8sClient.Status().Update(ctx, vmwcreds); err != nil {
+            return nil, fmt.Errorf("failed to reset VMwareCreds status: %w", err)
+        }
+
+    case "OpenstackCreds":
+        oscreds := &vjailbreakv1alpha1.OpenstackCreds{}
+        if err := p.K8sClient.Get(ctx, k8stypes.NamespacedName{Name: name, Namespace: namespace}, oscreds); err != nil {
+            return nil, fmt.Errorf("failed to get OpenstackCreds %s/%s: %w", namespace, name, err)
+        }
+
+        oscreds.Status.OpenStackValidationStatus = ""
+        oscreds.Status.OpenStackValidationMessage = ""
+        if err := p.K8sClient.Status().Update(ctx, oscreds); err != nil {
+            return nil, fmt.Errorf("failed to reset OpenstackCreds status: %w", err)
+        }
+
+    default:
+        return nil, fmt.Errorf("unknown credentials kind: %s", kind)
+    }
+
+    return &api.RevalidateCredentialsResponse{
+        Message: fmt.Sprintf("Re-validation triggered for %s %s/%s", kind, namespace, name),
+    }, nil
+}
```

Comment:
we are still relying on the controller to do the validation for us. It is not the expected behaviour of this PR. What we need is the actual re-validation logic here in the proxy code